### ISSUE TITLE
Prompt optimisation for self-recognition experiments

### ISSUE THREAD

**Issue by NelsonG-C** [NelsonG-C/lasr-labs-2025-control-project#89]
We need to move beyond our simple prompts and try some prompt engineering for self recognition to get an optimised prompt.

Deliverable:
Have the prompt we want to use for the general runs here

**Comment by Barahlush** [comment]
Implementing prompt optimization/evals in the `prompt-optimization` branch.

Using prompt optimization algorithms from DSPy, also try to use prompt improvement service in [Anthropic Console Workbench](https://console.anthropic.com/workbench/31a4a4da-523e-45c6-b3a0-0cb68d1ddf65?tab=prompt).

To optimize prompts I generated a bunch of training data using APPS training split (first 15 examples) using ['gpt-5-nano', 'human', 'gpt-4.1-nano', 'gemini-2.5-flash', 'o4-mini', 'gemini-2.5-flash-lite', 'gpt-4o-mini', 'claude-3-5-haiku-20241022'].

DSPy improved binary self-recognition accuracy in gpt-4o-mini from 52% to 62%. 

Plan to eval more models and also try Anthropic improved prompt:
- [ ] Write a script to evaluate arbitrary prompts on provided models
- [ ] Eval prompt from Anthropic Console
- [ ] Ask ChatGPT/Gemini/Anthropic frontier models to come up with a prompt, eval it
- [ ] Eval DSPy-optimized for gpt-4o-mini prompt on other models
- [ ] Try to DSPy-optimize prompts for other models

**Comment by Barahlush** [comment]
Okay, deprioritizing this for now, switching to fine-tuning after the meeting with Charlie. Seems like fine-tuning will be much more predictable and easy to do.
